mm2src/lp_ordermatch.rs

```diff
@@ -2021,31 +1903,38 @@ impl Orderbook {
     fn process_keep_alive(
         &mut self,
         from_pubkey: &str,
-        pairs: Vec<String>,
         message: new_protocol::PubkeyKeepAlive,
     ) -> Option<OrdermatchRequest> {
         let pubkey_state = pubkey_state_mut(&mut self.pubkeys_state, from_pubkey);
-        for pair in pairs {
-            if !pubkey_state.order_pairs_trie_roots.contains_key(&pair)
-                && self
-                    .topics_subscribed_to
-                    .contains_key(&orderbook_topic_from_ordered_pair(&pair))
+
+        let mut trie_roots_to_request = HashMap::new();
+        for (alb_pair, trie_root) in message.trie_roots {
+            if !self
```

Comment:
I've noticed that in case of relay node `topics_subscribed_to` might be empty, but node is still subscribed to all topics. Could you please pass `i_am_relay` here somehow and modify this condition to `!self.topics_subscribed_to && !i_am_relay`?